improver_tests/utilities/test_neighbourhood_tools.py

```diff
@@ -56,16 +57,34 @@ def test_neighbourhood_size_2(self):
 
     def test_exception_too_many_dims(self):
         """Test an exception is raised if shape has too many dimensions."""
-        msg = "Number of dimensions of input_array must be greater than or equal to length of shape"
+        msg = (
+            "Number of dimensions of the input array must be greater than or equal to "
+            "the length of the neighbourhood shape used for constructing rolling window neighbourhoods"
+        )
         with self.assertRaisesRegex(ValueError, msg):
             rolling_window(self.array, (2, 2, 2))
 
     def test_exception_dims_too_large(self):
-        """Test an exception is raised if dimensions of shape are larger than corresponding dimensions of input array."""
-        msg = "New shape contains a dimension that is negative or zero"
+        """Test an exception is raised if dimensions of shape are larger than 
+        corresponding dimensions of input array."""
+        msg = (
+            "The calculated shape of the output array view contains a dimension that is negative or zero. "
+            "Each dimension of the neighbourhood shape must be less than or equal to the corresponding "
```

Comment:
It might be worth reducing these line lengths to be closer to 80, and for the other error message, just for aesthetics and to avoid code-folding in a browser. [Black](https://black.readthedocs.io/en/stable/index.html) seems to be ignoring the line lengths of these strings, although it would have auto-formatted lines of this length if they were code.

Reply:
Done

Reply:
On this point: Black won't break strings even if they are on lines longer than the Black default of 88 characters. So this must be done manually if we wish to keep to the 88 character default width.

In addition, docstrings should be on lines no longer than 72 characters (including the leftmost white space) according to PEP8. Again, Black won't reformat to ensure this, so it must be done manually (as far as I'm aware).